### DISCUSSION TITLE
Using Maxwell-Boltzmann distribution with spatially varying temperature

### DISCUSSION THREAD

**Discussion by iamcalvinlau** [BLAST-WarpX/warpx#2023] [Simulation Inputs] [answered]
Hi, I am looking into using WarpX for a simulation with initialization based on an equilibrium from a different code. 

For initializing the magnetic field, I've side-stepped modifying the source code by decomposing the equilibrium fields via FFT and taking advantage of the maths parser. I can also do the same for the densities.

However, for the velocity distributions, I'm not really sure if there is any easy way to use the MB distribution with a spatially varying temperature in WarpX as it is right now. I've tried to search the git repo to see if anyone else has mentioned this and found no comments. 

Anyone have a suggestion on getting around this (maybe I've missed something in the documentation..) or, very hopefully, has anyone got a modified version lying around for this particular feature? Unfortunately, I'm mostly familiar with Fortran and not C++ so it'd take me some time to get up to speed for any modification at all..

Thanks,
Calvin

**Comment by ax3l** [comment]
Hi,

Thanks for the suggestion! This should not be too hard to add, we just need to pass `.theta` as a function of (x,y,z) for this initialization.

It's not currently on our agenda, but I can put it on the developer meeting to either find a volunteer or a volunteer to guide you on which aspects of the code to change.

**Reply by iamcalvinlau** [reply]
Sounds great! I'd love to learn a bit more about how to do this particular modification as I have a couple other ideas I'd like to explore as well which may benefit from knowing how to set up custom functions in the input. 

Thanks in advance!

**Reply by ax3l** [reply, marked as answer]
Hi, @hklion took this task on and we are soon merging the update via https://github.com/ECP-WarpX/WarpX/pull/2357

**Reply by iamcalvinlau** [reply]
Thanks @ax3l and @hklion!
